frontend/app/assets/javascripts/locations.location_holdings.js.erb

```diff
@@ -22,20 +23,68 @@ $(function () {
     });
   };
 
-  $('.location-holdings').each(init_locationHoldingsSearch);
-  $(document).bind('loadedrecordform.aspace', function (event, $container) {
-    $('.location-holdings', $container).each(init_locationHoldingsSearch);
+  // This bit works for a regular index page
+  // (the whole page reloads, so no MutationObserver is necessary)
+  $('.location-holdings').each(getLocationHoldingsCount);
+  $(document).on('loadedrecordform.aspace', function (event, $container) {
+    $('.location-holdings', $container).each(getLocationHoldingsCount);
   });
 
-  // GH-1920, listen for Browse Locations modal
-  var $locationBrowseBtn = $('input[data-label="Location"]')
-    .siblings()
-    .find('.linker-browse-btn');
+  // allows us to use await/then to wait for an element with the selector to appear
+  async function waitForElement(selector) {
+    return new Promise(resolve => {
+      // if already existing, just resolve
+      var el = $(selector);
+      if (el.length) {
+          return resolve(el);
+      }
+      const observer = new MutationObserver(mutations => {
+        var el = $(selector);
```

Comment:
Best to make both `el` variables `const` rather than `var` here.

Reply:
Indeed - can't recall why I used `var` here